### ISSUE TITLE
Wrapped line writes over prompt and beginning of line

### ISSUE THREAD

**Issue by bisen2** [thestinger/termite#601]
Hey, just wanted to say to start off, this is a great program and I am really loving it. I have noticed an odd bug: when a command that is long enough to need to wrap is typed in, the part that should be on the next line writes over the beginning of this line. Below are a few pictures demonstrating what I mean:
![first](https://user-images.githubusercontent.com/39412411/40634501-939eb208-62b2-11e8-9c88-fe50b6900360.png)
![sceond](https://user-images.githubusercontent.com/39412411/40634507-9b568cd2-62b2-11e8-87a8-ae522c8f9dfc.png)
![third](https://user-images.githubusercontent.com/39412411/40634511-9e38d338-62b2-11e8-9f2d-4dac4a2ef877.png)
![fourth](https://user-images.githubusercontent.com/39412411/40634512-a1669cca-62b2-11e8-84e9-ad64c0a17f37.png)
![fifth](https://user-images.githubusercontent.com/39412411/40634517-a4a520aa-62b2-11e8-96ac-3ef08e061019.png)
Then, if you begin to backspace the command, the remaining part of the overwritten line disappears:
![sixth](https://user-images.githubusercontent.com/39412411/40634519-a7986d58-62b2-11e8-8462-601679a160aa.png)
![seventh](https://user-images.githubusercontent.com/39412411/40634525-aa60e646-62b2-11e8-83a5-ad29dde144d6.png)
![eighth](https://user-images.githubusercontent.com/39412411/40634530-afb19a14-62b2-11e8-9160-9ce238dcc7bc.png)
If you backspace as far as possible, the cursor ends at a distance equivalent to the length of the promp plus the length of characters put on the "second" line. At this point, any command can be typed and will act as if the prompt were right in front of this. Also, if enough characters are typed to force a second wrapping, that goes exactly as expected, forming a new line below the previous.

I am running termite 13-3 on Arch 4.16.11 (latest at time of writing) and have also installed termite-terminfo:
![version](https://user-images.githubusercontent.com/39412411/40634642-442055aa-62b3-11e8-946d-0b9911a58f9e.png)

**Comment by egmontkob** [comment]
See under the PROMPTING section in bash's manual: non-printable characters in your PS1 need to be enclosed between `\[` and `\]`.

**Comment by bisen2** [comment]
Thanks! Sorry for the noob mistake

**Comment by Aeres-u99** [comment]
care to explain a bit more? I am kinda lost as to how to fix it.

**Comment by bisen2** [comment]
Essentially, your prompt (probably set as `PS1=...` in your .bashrc), includes some non printable characters, and in order for it to render properly, you need to enclose them in a set of \[ and \]. For example, the prompt shown here 
![image](https://user-images.githubusercontent.com/39412411/68549567-72f11980-03c7-11ea-8a81-366429cba773.png)
can be made and works properly with `PS1='\[\e[34m\]\u@\h\[\e[m\] \[\e[33m\]\W\[\e[m\] \$ '`. I was trying to color my `<<user>>@<<host>>` (created by the `\u@\h` expression) by enclosing them in a `\e[34m` and `\e[m` set, which colors everything inside the set with the color described by the number 34. But, the characters `\e[34m` and `\e[m` are not printable characters, so they cause issues if you just stick them into `PS1`. By enclosing every instance of these non-printable characters in a `\[` and `\]` set, you can make sure that they render properly and do not interfere with the operation of your prompt.

TL;DR:
If you enclose non-printable characters with `\[` and `\]`, it should fix any issues you are having.

**Comment by Aeres-u99** [comment]
Thank you for explanation! Truly appreciated
